OpenRA.Mods.Common/Traits/World/TerrainTunnelLayer.cs

```diff
@@ -0,0 +1,94 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class TerrainTunnelLayerInfo : ITraitInfo, Requires<DomainIndexInfo>, ILobbyCustomRulesIgnore
+	{
+		[Desc("Terrain type used by cells outside any tunnel footprint.")]
+		public readonly string ImpassibleTerrainType = "Impassable";
```

Comment:
I'm not a native english speaker, but using Impass**i**ble as field name but "Impassable" for the terrain type at least looks inconsistent.